tutorials/dataframe/df103_NanoAODHiggsAnalysis.py

```diff
@@ -0,0 +1,409 @@
+## \macro_code
+## \macro_output
+##
+## \date June 2019
+## \author Stefan Wunsch (KIT, CERN), Vincenzo Eduardo Padulano (UniMiB, CERN)
+
+import ROOT
+
+# Enable multi-threading
+ROOT.ROOT.EnableImplicitMT()
+
+# Include necessary header
+higgs_header_path = str(ROOT.gROOT.GetTutorialDir())
+higgs_header_path += "/dataframe/df103_NanoAODHiggsAnalysis.h"
+ROOT.gInterpreter.Declare('#include "{}"'.format(higgs_header_path))
+
+# Python functions
+def reco_higgs_to_2el2mu(df):
+    """Reconstruct Higgs from two electrons and two muons"""
+    # Filter interesting events
+    df_base = selection_2el2mu(df)
+    # Compute masses of Z systems
+    df_z_mass = df_base.Define(
+    "Z_mass",
+    "compute_z_masses_2el2mu(Electron_pt, Electron_eta, Electron_phi, Electron_mass, Muon_pt, Muon_eta, Muon_phi, Muon_mass)"
+    )
+    # Cut on mass of Z candidates
+    df_z_cut = filter_z_candidates(df_z_mass)
+    # Reconstruct H mass
+    df_h_mass = df_z_cut.Define(
+    "H_mass",
+    "compute_higgs_mass_2el2mu(Electron_pt, Electron_eta, Electron_phi, Electron_mass, Muon_pt, Muon_eta, Muon_phi, Muon_mass)"
+    )
+
+    return df_h_mass
+
+def selection_2el2mu(df):
+    """Select interesting events with two electrons and two muons"""
+    df_ge2el2mu = df.Filter("nElectron>=2 && nMuon>=2", "At least two electrons and two muons")
+    df_eta = df_ge2el2mu.Filter("All(abs(Electron_eta)<2.5) && All(abs(Muon_eta)<2.4)", "Eta cuts")
+
+    df_pt = df_eta.Filter("pt_cuts(Muon_pt, Electron_pt)", "Pt cuts")
+
+    df_dr = df_pt.Filter("dr_cuts(Muon_eta, Muon_phi, Electron_eta, Electron_phi)", "Dr cuts")
+
+    df_iso = df_dr.Filter("All(abs(Electron_pfRelIso03_all)<0.40) && All(abs(Muon_pfRelIso04_all)<0.40)",
+                              "Require good isolation")
+    df_el_ip3d = df_iso.Define("Electron_ip3d_el", "sqrt(Electron_dxy*Electron_dxy + Electron_dz*Electron_dz)")
+    df_el_sip3d = df_el_ip3d.Define("Electron_sip3d_el",
+                                        "Electron_ip3d_el/sqrt(Electron_dxyErr*Electron_dxyErr + "
+                                        "Electron_dzErr*Electron_dzErr)")
+    df_el_track = df_el_sip3d.Filter("All(Electron_sip3d_el<4) && All(abs(Electron_dxy)<0.5) && All(abs(Electron_dz)<1.0)",
+                                         "Electron track close to primary vertex with small uncertainty")
+    df_mu_ip3d = df_el_track.Define("Muon_ip3d_mu", "sqrt(Muon_dxy*Muon_dxy + Muon_dz*Muon_dz)")
+
+    df_mu_sip3d = df_mu_ip3d.Define(
+        "Muon_sip3d_mu",
+        "Muon_ip3d_mu/sqrt(Muon_dxyErr*Muon_dxyErr + Muon_dzErr*Muon_dzErr)"
+    )
+    df_mu_track = df_mu_sip3d.Filter("All(Muon_sip3d_mu<4) && All(abs(Muon_dxy)<0.5) && All(abs(Muon_dz)<1.0)",
+                                         "Muon track close to primary vertex with small uncertainty")
+    df_2p2n = df_mu_track.Filter("Sum(Electron_charge)==0 && Sum(Muon_charge)==0",
+                                     "Two opposite charged electron and muon pairs")
+
```

Comment:
is the formatting above ok? I don't think you need extra indentation for aguments on new lines, it makes it harder to read

Reply:
Missed that, sorry! Is there any set of rules for Python coding style in ROOT? Otherwise, I will change the tutorial to be [PEP8](https://www.python.org/dev/peps/pep-0008/)-ish compliant.

Reply:
PEP8-ish is fine :smile: that's the coding style